src/asm/parser.y

```diff
@@ -421,27 +424,45 @@ endc		: T_POP_ENDC T_NEWLINE {
 		}
 ;
 
-scoped_id	: T_ID | T_LOCAL_ID ;
+scoped_id	: T_ID
+			| T_LOCAL_ID
+			| T_OP_EVAL T_LPAREN string T_RPAREN {
+				struct Symbol const *sym = sym_FindScopedSymbol($3);
+
+				if (!sym)
+					fatalerror("EVAL: Symbol does not exist '%s'\n", $3);
```

Comment:
Then `EVAL` couldn't be used to forward-reference a symbol? (`ld a, ExportedSomewhereElse` works, so this should too, shouldn't it?)

Reply:
You're right, there's no need to look up the symbol `$3` and use its name; `$3` *is* the name.